bin/test-storybook.js

```diff
@@ -166,9 +168,15 @@ const main = async () => {
 
   process.env.STORYBOOK_CONFIG_DIR = runnerOptions.configDir;
 
-  const { storiesPaths } = getStorybookMetadata();
+  const { storiesPaths, lazyCompilation } = getStorybookMetadata();
   process.env.STORYBOOK_STORIES_PATTERN = storiesPaths;
 
+  if (lazyCompilation && isLocalStorybookIp) {
+    log(
+      'Detected a lazy-compiled Storybook. This will likely cause issues; consider disabling `lazyCompilation` for `test-storybook`.'
```

Comment:
We might want to explain a bit more what disabling lazyCompilation for test-storybook means. What about this?

```suggestion
      `You're running Storybook with lazy compilation enabled, and will likely cause issues with the test runner locally. Consider disabling 'lazyCompilation' in main.js when running 'test-storybook' locally.`
```